crates/polars-ops/src/series/ops/horizontal.rs

```diff
@@ -78,3 +79,10 @@ pub fn min_horizontal(s: &[Series]) -> PolarsResult<Option<Series>> {
     let df = DataFrame::new_no_checks(Vec::from(s));
     df.hmin().map(|opt_s| opt_s.map(|s| s.with_name("min")))
 }
+
+#[cfg(feature = "zip_with")]
+pub fn mean_horizontal(s: &[Series], none_strategy: NullStrategy) -> PolarsResult<Option<Series>> {
+    let df = DataFrame::new_no_checks(Vec::from(s));
+    df.hmean(none_strategy)
+        .map(|opt_s| opt_s.map(|s| s.with_name("mean")))
```

Comment:
We don't need to rename the output, but instead use the first name (left-most series) as output name. I think we should correct that for the `min` and `max` cases as well.

Reply:
Not sure if i completely understand, but that would mean that if we do:

```
df = pl.DataFrame(
    {
        "a": [1, 8, 3],
        "b": [None, 7, 8],
    }
)


df.with_columns(pl.mean_horizontal("a", "b"))
```
```
# take leftmost name will lead to that column containing the mean values
shape: (3, 2)
┌─────┬──────┐
│ a   ┆ b    │
│ --- ┆ ---  │
│ f64 ┆ i64  │
╞═════╪══════╡
│ 1.0 ┆ null │
│ 7.5 ┆ 7    │
│ 5.5 ┆ 8    │
└─────┴──────┘

instead of 

# mean is a separate column
shape: (3, 3)
┌─────┬──────┬──────┐
│ a   ┆ b    ┆ mean │
│ --- ┆ ---  ┆ ---  │
│ i64 ┆ i64  ┆ f64  │
╞═════╪══════╪══════╡
│ 1   ┆ null ┆ 1.0  │
│ 8   ┆ 7    ┆ 7.5  │
│ 3   ┆ 8    ┆ 5.5  │
└─────┴──────┴──────┘
```

I would expect we would want the second option.

Reply:
We should not give expressions names. You should do that.

So that would be `df.with_columns(mean = pl.mean_horizontal("a", "b"))`

Reply:
Ah yes I understand. Changing it for `min` and `max` could be a breaking change right? The `min_horizontal` example in the documentation shows:

```
df = pl.DataFrame(
    {
        "a": [1, 8, 3],
        "b": [4, 5, None],
        "c": ["x", "y", "z"],
    }
)
df.with_columns(pl.min_horizontal("a", "b"))
shape: (3, 4)
┌─────┬──────┬─────┬─────┐
│ a   ┆ b    ┆ c   ┆ min │
│ --- ┆ ---  ┆ --- ┆ --- │
│ i64 ┆ i64  ┆ str ┆ i64 │
╞═════╪══════╪═════╪═════╡
│ 1   ┆ 4    ┆ x   ┆ 1   │
│ 8   ┆ 5    ┆ y   ┆ 5   │
│ 3   ┆ null ┆ z   ┆ 3   │
└─────┴──────┴─────┴─────┘
```

meaning if people have it implemented like this, and we remove the renaming of the output, the min column will disappear and value will overwrite the values of the leftmost column.

Reply:
It was a bug in our implementation. You can fix it and update the docstrings.

Reply:
Do we want to adjust it also for "sum_horizontal, any_horizontal, all_horizontal"?

Also, why does the rust `sum_horizontal` implementation, not use `hsum`, but a custom implementation?

Reply:
Yes, we want to adjust that.

I am not sure why we don't use hsum. Does rewriting it work for all tests?

Reply:
Changed it, it works for all tests. Rebase coming up